android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveydetail/SurveyDetailViewModel.kt

```diff
@@ -0,0 +1,15 @@
+package co.nimblehq.ic.kmm.suv.android.ui.screens.surveydetail
+
+import co.nimblehq.ic.kmm.suv.android.ui.base.BaseViewModel
+import co.nimblehq.ic.kmm.suv.android.ui.screens.home.SurveyArgument
+
+class SurveyDetailViewModel(
+    val survey: SurveyArgument
```

Comment:
🤔 I don't think we should put the argument as the ViewModel's dependencies. The argument can be passed into the View layer rather than ViewModel layer. Could you please kindly check the implementation in Compose Crypto from [here](https://github.com/nimblehq/jetpack-compose-crypto/blob/939fff5c148c4ede2d3a875db10eb74deb946211/app/src/main/java/co/nimblehq/compose/crypto/ui/navigation/AppNavigation.kt#L24-L29)

Reply:
Fixed in: https://github.com/suho/kmm-ic/pull/112/commits/2510eab5048f645017fb6c8e5dd19f801b627b91

Reply:
Another good thing when I research about this, we can use [SavedStateHandle](https://developer.android.com/reference/androidx/lifecycle/SavedStateHandle), but I will not implement in this PR, will improve later :D

Reply:
@suho Yeah. The `SavedStateHandle` is the approach for handling the result returns from the detail screen to the previous screen. 

https://developer.android.com/guide/navigation/navigation-programmatic#returning_a_result